### ISSUE TITLE
Plugin error colour scheme is hard to read

### ISSUE THREAD

**Issue by DragaDoncila** [napari/napari#2257]
## 🐛 Bug

The colour scheme for displaying plugin errors is difficult to read for both light and dark themes.

![errors_light](https://user-images.githubusercontent.com/17995243/107724769-c2e02e00-6d38-11eb-9240-c37dc13de54e.png) ![errors_dark](https://user-images.githubusercontent.com/17995243/107724778-c70c4b80-6d38-11eb-97f9-0d182e7ae939.png)

This may require some input on design from the design team notably for accessibility considerations.

## Additional context

To look at fixing this @tlambert03 recommends starting in [napari.plugins.exceptions](https://github.com/napari/napari/blob/master/napari/plugins/exceptions.py)

**Comment by tlambert03** [comment]
@DragaDoncila, if you want to play with this, I think you should just be able to  change some of the colors in the `ANSI_STYLES` dict in that file.  it's generating html, so any valid css "color" style will work.  you might look to [pygments](https://pygments.org/demo/) for inspiration.

**Comment by tlambert03** [comment]
actual pygments colors are here: https://github.com/pygments/pygments/tree/master/pygments/styles

**Comment by tlambert03** [comment]
(bonus points for figuring out how to make it change with the theme :joy:)

**Comment by DragaDoncila** [comment]
> (bonus points for figuring out how to make it change with the theme :joy:)

"left as an exercise to the reader" lol

**Comment by isabela-pf** [comment]
I’m always happy to see issues opened that cover accessibility! Here are some thoughts on what these colors lack and what they could be updated to.

## Current state

I’m still new to napari, so please let me know if this information sounds incorrect or incomplete. 

- There are 8 colors (and an additional background color) that the console currently uses.
- Light and dark mode does influence the background color of the console, but not the syntax highlighting colors. Dark theme uses `#000000` and light theme uses `#FFFFFF`.
- Very few of the current colors pass any [accessible color contrast standards](https://www.w3.org/TR/WCAG22/#contrast-minimum). I’ve noted the status in the image list below.

This is what the current colors look like as a list:

| Color name | Hex code | Contrast in dark theme | Contrast in light theme |
| --- | --- | --- | --- |
| Black | `#000000` | Fail | AAA |
| Red | `#FF0000` | AA | Fail |
| Green | `#008000` | Fail | AA |
| Yellow | `#FFFF00` | AAA | Fail |
| Blue | `#0000FF` | Fail | AAA |
| Magenta | `#FF00FF` | AA | Fail |
| Cyan | `#00FFFF` | AAA | Fail |
| White | `#FFFFFF` | AAA | Fail |

## Proposals

I’d really advocate for having the colors (all, not just the background) switch with napari’s theme. Finding adequate contrast with a single set of colors on two different extremes will be very limiting if it is even possible.

### Match napari hub syntax highlighting

@liaprins-czi told me that the theming for napari hub included a proposal for syntax highlighting with accessible contrast. That’s great! I don’t see a reason why we wouldn’t want to use that work in the napari viewer as well. 

I find it helpful for users to have consistency even across related interfaces. This could also mean less maintenance for the project since we won’t have to deal with multiple themes and/or find multiple solutions to the same problem.

Because napari hub uses a light palette, the below option for the light theme is the same as @liaprins-czi’s work while the dark theme variant is my interpretation extending it.
#### Light theme

Background: `#F7F7F7`

| Hex code | Contrast in light theme |
| --- | --- |
| `#000000` | AAA |
| `#6F6F6F` | AA |
| `#0074B8` | AA |
| `#956441` | AA |
| `#227B81` | AA |
| `#7518A1` | AAA |
| `#004166` | AAA |
| `#5D7A1F` | AA |

Here’s roughly how that looks (not shown in the console):
<img width="934" alt="Example (0-1)" src="https://user-images.githubusercontent.com/50221806/134701922-cf2ef322-d634-4aa6-a8b3-d3512dff328e.png">

#### Dark theme

Background: `#000000`

| Hex code | Contrast in dark theme |
| --- | --- |
| `#F7F7F7` | AAA |
| `#9A9A9A` | AAA |
| `#82DCF0` | AAA |
| `#BF8D68` | AAA |
| `#55B69E` | AAA |
| `#B780D1` | AAA |
| `#4CA0ED` | AAA |
| `#A1C25B` | AAA |

Rough example:

<img width="934" alt="Example (0-6)" src="https://user-images.githubusercontent.com/50221806/134702069-79e4f117-1f22-4ef5-8e5c-a2afd72392c5.png">

### Update syntax highlighting based on current colors

If there is a reason that we want to stick with the same set of colors currently being used, I’ve made some sets based on the same colors with different lightness depending on the background. 

#### Light theme
Background: `#FFFFFF`

| Hex code | Contrast in light theme |
| --- | --- |
| `#6F6F6F` | AA |
| `#A41313` | AAA |
| `#087708` | AA |
| `#AD5401` | AA |
| `#3333AC` | AAA |
| `#A41E96` | AA |
| `#0E7E7E` | AA |
| `#000000` | AAA |

Rough example:
<img width="934" alt="Example (0-7)" src="https://user-images.githubusercontent.com/50221806/134702011-914405c3-dcea-4b1f-875c-4ce1b0ed0494.png">

#### Dark theme

Background: `#000000`

| Hex code | Contrast in dark theme |
| --- | --- |
| `#959595` | AAA |
| `#ED7979` | AAA |
| `#73B973` | AAA |
| `#E9C057` | AAA |
| `#8E8EED` | AAA |
| `#F180F1` | AAA |
| `#7CC6C6` | AAA |
| `#F4F1F1` | AAA |

Rough example:
<img width="934" alt="Example (0-4)" src="https://user-images.githubusercontent.com/50221806/134701992-71f41fd4-82ce-4306-9afe-0d8215ba30ad.png">


What are your thoughts? Comments? Concerns?

**Comment by goanpeca** [comment]
>I’d really advocate for having the colors (all, not just the background) switch with napari’s theme. Finding adequate contrast with a single set of colors on two different extremes will be very limiting if it is even possible.

Sounds good @isabela-pf !

Just wondering if every new theme will need to also update theses colors? I gues that would be the case right?

**Comment by DragaDoncila** [comment]
@isabela-pf thanks for looking into this, they are beautiful! I tend to agree that it's going to be difficult to share one scheme across the dark and light themes.

I absolutely love the dark theme colours you've come up with I think they're so cute! But I also agree that consistency for users is very important and would therefore rather lean towards using the napari hub/napari.org colours in our viewer as well.

**Comment by isabela-pf** [comment]
@goanpeca by "every new theme" do you mean themes added in the future or via plugins? I'm asking because I only saw light and dark themes in default napari and so that's all I was accounting for so far.

If there are other themes, the most important ones to keep meeting contrast standards are the default ones napari viewer proper maintains. While I'm always going to advocate for it, other themes may have reason to not meet those contrast standards. After all, contrast isn't the only way something visual can or can't be accessible.

**Comment by goanpeca** [comment]
I think we can now add themes via plugins, so thinking of those. I think we could offer some checks on this values to guide users. Default themes in napari are indeed those 2 alone.

Will think of how to pro ide better feedback and do some checks on the colors :)

Thanks!

**Comment by Czaki** [comment]
it looks solved by #4414
